### DISCUSSION TITLE
SchemaGenerator and path parameter converters

### DISCUSSION THREAD

**Discussion by jvail** [encode/starlette#1645] [Potential Issue] [answered]
Hi,

if I use type converters in a route definition ('/somepath/{parameter:int}') and generate an openAPI schema then I'd expect that the type, the converter is stripped from the URL template. But currently i get 

```json
"paths": {
        "//somepath/{parameter:int}": {}
}
```

instead of 

```json
"paths": {
        "//somepath/{parameter}": {}
}
```

Is that by purpose?
If so the parameter names inside the doc string don't match anymore if parsed e.g. inside postman. Postman would create a parameter named "parameter:int" instead of just "parameter".

Thank you
Jan

**Comment by Kludex** [comment, marked as answer]
How did you generate the `openapi.json`?

**Reply by jvail** [reply]
I am returning a JSONResponse with `schemas.get_schema(routes=app.routes)`  because I wanted to return JSON instead of yml - worng?

I create the schema obj like this 

```python
schemas = SchemaGenerator(
    {"openapi": "3.1.0", "info": {"title": "Example API", "version": "1.0"}}
)
```

**Reply by Kludex** [reply]
Thanks for the report. Issue created: https://github.com/encode/starlette/issues/1646
